hudi-common/src/main/java/org/apache/hudi/common/util/HoodieCommonKryoRegistrar.java

```diff
@@ -85,7 +96,19 @@ public void registerClasses(Kryo kryo) {
 
         HoodieRecordLocation.class,
         HoodieRecordGlobalLocation.class,
-        HoodieRecordDelegate.class
+        HoodieRecordDelegate.class,
+        HoodieInstant.class,
```

Comment:
what is the criteria for adding these classes?

Reply:
Kryo recommends registering the classes since it uses a shorter identifier instead of the fully qualified class name when serializing objects. I don't have any context for what the criteria is within the Hudi repo though.